torch_geometric/nn/kge/rotate.py

```diff
@@ -0,0 +1,92 @@
+import math
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+from torch.nn import Embedding
+
+from torch_geometric.nn.kge import KGEModel
+
+
+class RotatE(KGEModel):
+    r"""The RotatE model from the `"RotatE: Knowledge Graph Embedding by
+    Relational Rotation in Complex Space" <https://arxiv.org/abs/
+    1902.10197>`_ paper.
+
+    :class:`RotatE` models relations as a rotation in complex space
+    from head to tail. The scoring function is thus:
+
+    .. math::
+        d(h, r, t) = - {\| \mathbf{e}_h \circ \mathbf{e}_r - \mathbf{e}_t \|}_p
+
+    .. note::
+
+        For an example of using the :class:`RotatE` model, see
+        `examples/kge_fb15k_237.py
+        <https://github.com/pyg-team/pytorch_geometric/blob/master/examples/
+        kge_fb15k_237.py>`_.
+
+    Args:
+        num_nodes (int): The number of nodes/entities in the graph.
+        num_relations (int): The number of relations in the graph.
+        hidden_channels (int): The hidden embedding size.
+        margin (float, optional): The margin of the ranking loss.
+        sparse (bool, optional): If set to :obj:`True`, gradients w.r.t. to
+            the embedding matrices will be sparse. (default: :obj:`False`)
+    """
+    def __init__(
+        self,
+        num_nodes: int,
+        num_relations: int,
+        hidden_channels: int,
+        margin: float = 1.0,
+        sparse: bool = False,
+    ):
+        super().__init__(num_nodes, num_relations, hidden_channels, sparse)
+        self.margin = margin
```

Comment:
```suggestion
        self.register_buffer('margin', margin)
```
This way `self.margin` will be on the same `device` as other parameters.

Reply:
Done, just an FYI had to wrap margin in `torch.Tensor`, wanted to make sure thats ok